src/Castle.Core/DynamicProxy/ProxyUtil.cs

```diff
@@ -26,11 +26,84 @@ namespace Castle.DynamicProxy
 #endif
 
 	using Castle.Core.Internal;
+	using Castle.DynamicProxy.Generators;
+	using Castle.DynamicProxy.Internal;
 
 	public static class ProxyUtil
 	{
 		private static readonly SynchronizedDictionary<Assembly, bool> internalsVisibleToDynamicProxy = new SynchronizedDictionary<Assembly, bool>();
 
+		/// <summary>
+		///   Attempts to create a delegate of the specified type to a suitable `Invoke` method
+		///   on the given <paramref name="proxy"/> instance.
+		/// </summary>
+		/// <param name="proxy">The proxy instance to which the delegate should be bound.</param>
+		/// <typeparam name="TDelegate">The type of delegate that should be created.</typeparam>
+		/// <param name="delegate">The produced delegate if the method succeeds; otherwise <see langword="null"/>.</param>
+		/// <returns><see langword="true"/> if the method succeeds; otherwise <see langword="false"/>.</returns>
+		public static bool TryCreateDelegateToMixin<TDelegate>(object proxy, out TDelegate @delegate)
+		{
+			if (TryCreateDelegateToMixin(proxy, typeof(TDelegate), out var d))
+			{
+				@delegate = (TDelegate)(object)d;
+				return true;
+			}
+			else
+			{
+				@delegate = default(TDelegate);
+				return false;
+			}
+		}
+
+		/// <summary>
+		///   Attempts to create a delegate of the specified type to a suitable `Invoke` method
+		///   on the given <paramref name="proxy"/> instance.
+		/// </summary>
+		/// <param name="proxy">The proxy instance to which the delegate should be bound.</param>
+		/// <param name="delegateType">The type of delegate that should be created.</param>
+		/// <param name="delegate">The produced delegate if the method succeeds; otherwise <see langword="null"/>.</param>
+		/// <returns><see langword="true"/> if the method succeeds; otherwise <see langword="false"/>.</returns>
+		public static bool TryCreateDelegateToMixin(object proxy, Type delegateType, out Delegate @delegate)
+		{
+			if (proxy == null)
```

Comment:
I thought our style guidelines allowed simple single line if-statements for control statements, throw, return, break, continue.

Reply:
Indeed they do! I changed a few instances of this to single-line formatting in https://github.com/castleproject/Core/pull/436/commits/68e8fefa55886a39a625f935a9d30aa355a3e1d0.